src/Orleans/Serialization/SerializationManager.cs

```diff
@@ -181,6 +187,7 @@ static SerializationManager()
             AppDomain.CurrentDomain.AssemblyResolve += OnResolveEventHandler;
 
             registeredTypes = new HashSet<Type>();
+            externalSerializers = new ConcurrentHashSet<IExternalSerializer>();
```

Comment:
Why do we need a `ConcurrentHashSet<T>` here? All access operations on it after initialization are read-only, right? Why not `List<T>` or `ImmutableList<T>` if there are concerns?